app/experimenter/experiments/api/v3/serializers.py

```diff
@@ -32,12 +32,17 @@ class Meta:
 
     def validate_name(self, name):
         slug = slugify(name)
-
-        if slug:
-            return name
-        raise serializers.ValidationError(
-            ["Name needs to contain alphanumeric characters"]
-        )
+        existing_slugs = Experiment.objects.values_list("slug", flat=True)
+
+        if not slug:
+            raise serializers.ValidationError(
+                ["Name needs to contain alphanumeric characters"]
+            )
+        if slug in existing_slugs:
+            raise serializers.ValidationError(
+                ["Name maps to a pre-existing slug, please choose another name"]
```

Comment:
We can just reuse the same error it already gives: `Experiment with this name already exists.`

Reply:
I wonder if there's a way to trigger that same error programmatically, but I don't see anything from a quick scan....